### PR TITLE
Combine all empty block rules into one multi rule - #95

### PR DESCRIPTION
**Pull request by arturbosch**

### CODE REVIEW

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/empty/EmptyBlocks.kt [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+			emptyWhileBlock
+	)
+
+	override fun visitKtFile(file: KtFile) {
+		super.visitKtFile(file)
+		report(rules.flatMap { it.findings })
+	}
+
+	override fun visitCatchSection(catchClause: KtCatchClause) {
+		if (emptyCatchBlock.active) {
```
Comment (Mauin) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/empty/EmptyBlocks.kt]: This is not checking if the rule is suppressed if I recall correctly.

Comment (arturbosch) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/empty/EmptyBlocks.kt]: The suppressing is done on report(findings) where I merge all findings :)

Comment (Mauin) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/empty/EmptyBlocks.kt]: Ah, alright.

Comment (arturbosch) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/empty/EmptyBlocks.kt]: But I should add a test case for this. Thx!

Comment (arturbosch) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/empty/EmptyBlocks.kt]: You were right. The `visitCondition` check does `isActive` AND `isSuppressedFileLevel` check. Updated.

**Comment by arturbosch** [comment]
Uff I introduced a lateinit :D. I don't quite like that filtering the active rules need a KtFile ... but I don't have a solution right now.

**Pull request merged**